### DISCUSSION TITLE
Security and Best Practices

### DISCUSSION THREAD

**Discussion by MM4MA** [pocketbase/pocketbase#5479] [Q&A]
Hi everyone,

I’m working on a small hobby project that started in school, and now I need to create a website for it. The website will be hosted on static Cloudflare Pages, and I plan to implement a login/register page.

I want to use PocketBase as the backend for this project. My plan is to add the JS SDK via the unpkg CDN and then implement a login/register functionality along with a small message storage feature (essentially a self-chat) that will be saved in PocketBase.

I have a couple of questions regarding security:

1. Does PocketBase handle SQL injection attempts, or do I need to take additional measures to prevent them?
2. What other security considerations should I be aware of when releasing a project with PocketBase?
Any advice or insights would be greatly appreciated!

Thanks in advance!

**Comment by ganigeorgiev** [comment]
If you are planning to use the JS SDK on the client-side (aka. from the browser) then you don't need to worry about SQL or string injection attacks. The PocketBase Web APIs convert the filter to proper SQL statements and non-identifiers are binded as parameters to the `database/sql` query execution. You should be more careful around loading untrusted 3rd party JS code and preventing XSS attacks (see also [JS SDK Security section](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#security)).

If you are planning to use PocketBase together with a frontend-meta framework in a JS SSR mode like SvelteKit, Nuxt, Next.js, etc., then you could check https://github.com/pocketbase/pocketbase/discussions/5313.

If you are planning to [use PocketBase as a framework](https://pocketbase.io/docs/use-as-framework/), then when executing db queries you'll need to bind untrusted user input as placeholder parameters on your own using the `{:paramName}` format. For more details, you could refer to https://pocketbase.io/docs/go-database/#binding-parameters.

In all cases you need to ensure that your [collections API rules](https://pocketbase.io/docs/api-rules-and-filters/) are properly set.

_Side-note: while PocketBase can be used as a backend for websites at the moment we don't have the necessary helpers for that and depending on your use case it may not be the best tool for the job. PocketBase was designed primarily for building applications that usually can be interact directly from the client-side (e.g. SPA in the browser or a mobile app)._
